src/types/Inputs.ts

```diff
@@ -28,11 +30,45 @@ export interface DefaultInputFieldDefinition {
   model?: InputFieldChoice[];
 }
 
-export interface CodeInputFieldDefinition extends DefaultInputFieldDefinition {
-  type: "code";
+/** Defines attributes of a InputField. */
+export interface DefaultInputFieldDefinition extends BaseInputFieldDefinition {
+  type: Exclude<InputFieldType, "code" | "connection">;
+}
+
+/** Defines attributes of a CodeInputField. */
+export interface CodeInputFieldDefinition extends BaseInputFieldDefinition {
+  type: Extract<InputFieldType, "code">;
   language?: string;
 }
 
+export enum OAuth2Type {
+  ClientCredentials = "client_credentials",
+  AuthorizationCode = "authorization_code",
+}
+
+/** Defines attributes of a ConnectionField. */
+export interface ConnectionFieldDefinition extends BaseInputFieldDefinition {
+  type: Extract<InputFieldType, "connection">;
+  connectionKey: string;
+  oauth2Type?: OAuth2Type;
+  iconPath?: string;
+  inputs: ConnectionInputs;
+}
+
+export interface Connection<
+  TField extends ConnectionFieldDefinition = ConnectionFieldDefinition
+> {
+  /** Key of the Connection type. */
+  key: TField["connectionKey"];
+  /** Identifier for the Config Variable hosting this Connection. */
+  instanceConfigVarId: string;
+  /** Field values supplied to this Connection. */
+  fields: { [Property in keyof TField["inputs"]]: unknown };
```

Comment:
Will a field value ever _not_ be a `string`? Should we type this `: string` instead of `: unkonwn`?